network-config-analyzer/ProtocolNameResolver.py

```diff
@@ -0,0 +1,94 @@
+                                     44: 'IPv6Frag', 45: 'IDRP', 46: 'RSVP', 47: 'GRE', 48: 'DSR', 49: 'BNA', 50: 'ESP',
+                                     51: 'AH', 52: 'INLSP', 53: 'SWIPEdeprecated', 54: 'NARP', 55: 'MOBILE', 56: 'TLSP',
+                                     57: 'SKIP', 58: 'ICMPv6', 59: 'IPv6NoNxt', 60: 'IPv6Opts', 62: 'CFTP',
+                                     64: 'SATEXPAK', 65: 'KRYPTOLAN', 66: 'RVD', 67: 'IPPC', 69: 'SATMON', 70: 'VISA',
+                                     71: 'IPCV', 72: 'CPNX', 73: 'CPHB', 74: 'WSN', 75: 'PVP', 76: 'BRSATMON',
+                                     77: 'SUNND', 78: 'WBMON', 79: 'WBEXPAK', 80: 'ISOIP', 81: 'VMTP', 82: 'SECUREVMTP',
+                                     83: 'VINES', 84: 'IPTM', 85: 'NSFNETIGP', 86: 'DGP', 87: 'TCF', 88: 'EIGRP',
+                                     89: 'OSPFIGP', 90: 'SpriteRPC', 91: 'LARP', 92: 'MTP', 93: 'AX25', 94: 'IPIP',
+                                     95: 'MICPdeprecated', 96: 'SCCSP', 97: 'ETHERIP', 98: 'ENCAP', 100: 'GMTP',
+                                     101: 'IFMP', 102: 'PNNI', 103: 'PIM', 104: 'ARIS', 105: 'SCPS', 106: 'QNX',
+                                     107: 'AN', 108: 'IPComp', 109: 'SNP', 110: 'CompaqPeer', 111: 'IPXinIP',
+                                     112: 'VRRP', 113: 'PGM', 115: 'L2TP', 116: 'DDX', 117: 'IATP', 118: 'STP',
+                                     119: 'SRP', 120: 'UTI', 121: 'SMP', 122: 'SMdeprecated', 123: 'PTP',
+                                     124: 'ISISoverIPv4', 125: 'FIRE', 126: 'CRTP', 127: 'CRUDP', 128: 'SSCOPMCE',
+                                     129: 'IPLT', 130: 'SPS', 131: 'PIPE', 132: 'SCTP', 133: 'FC', 134: 'RSVPE2EIGNORE',
+                                     135: 'MobilityHeader', 136: 'UDPLite', 137: 'MPLSinIP', 138: 'manet', 139: 'HIP',
+                                     140: 'Shim6', 141: 'WESP', 142: 'ROHC', 143: 'Ethernet'}
+
+    @staticmethod
+    def get_protocol_name(protocol_number: int) -> str:
+        """
+        :param protocol_number: Protocol number
+        :return: The protocol name
+        :rtype: str
+        """
+        if protocol_number < 1 or protocol_number > 255:
+            raise Exception('Protocol number must be in the range 1-255')
+
+        return ProtocolNameResolver._protocol_number_to_name_dict[protocol_number]
+
+    @staticmethod
+    def get_protocol_number(protocol_name: str) -> int:
+        """
+        :param protocol_name: Protocol name
+        :return: The protocol number
+        :rtype: int
+        """
+        if isinstance(protocol_name, int):
+            return protocol_name
+
+        protocol_num = ProtocolNameResolver._protocol_name_to_number_dict[protocol_name]
+        if not protocol_num:
+            raise Exception('Unknown protocol name: ' + protocol_name)
+
+        return protocol_num
+
+    @staticmethod
+    def is_protocol(protocol: int) -> bool:
```

Comment:
maybe rename to `is_standard_protocol` because protocol number larger than 143 may still be legitimate, user-defined protocols

Reply:
Done